src/main/kotlin/org/jetbrains/kotlinx/jupyter/repl/creating/creatingUtil.kt

```diff
@@ -37,7 +39,9 @@ fun createRepl(
         override fun provideIsEmbedded() = isEmbedded
         override fun provideDisplayHandler() = displayHandler
         override fun provideCommunicationFacility(): JupyterCommunicationFacility = communicationFacility
-        override fun provideDebugPort(): Int? = debugPort
+        override fun provideDebugIntegration(): DebugIntegrationProvider {
+            return NotebookDebugIntegrationProvider(debugPort)
```

Comment:
Once again, is there anything around this code that doesn't simply use passed constructor parameter?

Reply:
True, can be done. I'll change. However, I didn't think about this as a pattern to follow